src/common/src/cast/mod.rs

```diff
@@ -160,6 +176,40 @@ pub fn i64_to_timestamp(t: i64) -> Result<Timestamp> {
     ))
 }
 
+/// Converts str to interval for postgres debezium cdc json
+///
+/// The input str must have the following format:
+/// P<years>Y<months>M<days>DT<hours>H<minutes>M<seconds>S
+///
+/// Example
+/// - P1Y2M3DT4H5M6.78S
+#[inline]
+pub fn str_to_interval(s: &str) -> Result<Interval> {
```

Comment:
We can add this to common/src/types/interval.rs after the method `as_iso_8601`, and call this one `from_iso_8601`. This is an ISO standard, not only used by debezium.